frontend/src/pages/application/PipelineStage/form.vue

```diff
@@ -343,6 +343,16 @@ export default {
             return options
         }
     },
+    watch: {
+        'input.stageType': {
+            handler (_stageType) {
+                // if the stage type is changed, ensure the action is NOT "create_snapshot" when the stage type is "device"
+                if (this.input.stageType === StageType.DEVICE && this.input.action === 'create_snapshot') {
+                    this.input.action = 'use_latest_snapshot' // set default action for device stage
```

Comment:
I'd make this slightly more dynamic and pull the top option from the list, will make the change myself for speed!